src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py

```diff
@@ -36,6 +36,14 @@ def __init__(
             feature_extractor=feature_extractor,
         )
 
+    def set_attention_chunk(self, chunk_size: Optional[Union[str, int]] = "auto"):
+        # set chunk_size = `None` to disable `set_attention_chunk`
+        if chunk_size == "auto":
```

Comment:
Not sure about having it as a setter function, but adding another argument to `__call__` isn't ideal either... 
My reasoning is that we'll have multiple levels of optimization (fp16, accelerate loader, chunked softmax), and it would be great to have one place to control them all.

But maybe @patil-suraj or @pcuenca have a better idea where to put it API-wise, I'm just a bit conflicted about having a function here.

Reply:
Hmm - don't really see the problem here to be honest. Not that this is more or less exactly the same as `gradient_checkpointing_enable()` which is also a setter function (we trade speed for memory). `fp16` is also a setter function. Also I don't fully understand what you mean by "one place" for all?

Other options are:
- putting it in the config: no-go as it's not at all related to the architecture of the model
- forward argument to `forward()`: Don't like this
   - IMO parameters to the forward function should be limited to either tensors that are used or flags that change the type of output that is returned (e.g. `return_dict`, `output_type`). Optimization parameters should be setters IMO (like gradient checkpointing, `half()`, ....)

Reply:
Agree with Patrick here. The other options are aren't really ideal, so let's go with this.

Although, not sure if it's best keep this in the `pipeline`,  what if we do 
```python
pipe.unet.set_attention_chunk
```

as this specific to the model.

Reply:
I see the point but we don't have a general good default for all unets - it's a good default for stable diffusion so I think we need a pipeline function here